src/main/java/com/amuzil/omegasource/magus/network/packets/client_executed/SkillTriggeredPacket.java

```diff
@@ -0,0 +1,32 @@
+package com.amuzil.omegasource.magus.network.packets.client_executed;
+
+import com.amuzil.omegasource.magus.network.ClientPacketHandler;
+import com.amuzil.omegasource.magus.skill.skill.Skill;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraftforge.network.NetworkEvent;
+
+import java.util.function.Supplier;
+
+public class SkillTriggeredPacket {
+
+    private final Skill skill;
+
+    public SkillTriggeredPacket(Skill skill) {
+        this.skill = skill;
+    }
+
+    public void toBytes(FriendlyByteBuf buf) {
+        if(skill != null) {
+            //todo serialize skill
+        }
+    }
+
+    public static SkillTriggeredPacket fromBytes(FriendlyByteBuf buf) {
+        //todo deserialise skill
+        return new SkillTriggeredPacket(null);
```

Comment:
We do need a way to properly deserialise the Skill. You should be able to just grab the requisite Skill object from its forge registry by encoding and decoding the object's name. Ideally we post an event upon receiving this.